src/coreclr/utilcode/pedecoder.cpp

```diff
@@ -1774,18 +1773,22 @@ void PEDecoder::LayoutILOnly(void *base, BOOL allowFullPE) const
     for (section = sectionStart; section < sectionEnd; section++)
     {
         // Add appropriate page protection.
-#if defined(CROSSGEN_COMPILE) || defined(TARGET_UNIX)
-        if (section->Characteristics & IMAGE_SCN_MEM_WRITE)
-            continue;
+        DWORD newProtection;
```

Comment:
I would like to understand why this has changed for Unix and crossgen.

Reply:
We did not have a scenario for Unix that needed to make executable sections executable.
Since it was Windows-only scenario, so we could assume that anything Unix or crossgen does not need PAGE_EXECUTE_READ. 
Now we have a scenario for Unix too - when `enableExecution` parameter is passed.

Reply:
The reason why we did not need executable here on Unix is that we always could remap to the original data with executable protection. That does not work if the original data is compressed, and in such case we need to use the same approach as on Windows.